aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -185,6 +189,23 @@ module aptos_framework::code {
             request_publish(addr, module_names, code, policy.policy)
     }
 
+    public fun is_package_upgradeable(metadata_serialized: vector<u8>): bool {
+        let metadata = util::from_bytes<PackageMetadata>(metadata_serialized);
+        metadata.upgrade_policy.policy < upgrade_policy_immutable().policy
+    }
+
+    public fun freeze_package_registry(publisher: &signer, registry: Object<PackageRegistry>) acquires PackageRegistry {
+        let registry_addr = object::object_address(&registry);
+        assert!(exists<PackageRegistry>(registry_addr), error::not_found(EPACKAGE_DEP_MISSING));
+        assert!(object::is_owner(registry, signer::address_of(publisher)), ENOT_OWNER);
+
+        let registry = borrow_global_mut<PackageRegistry>(registry_addr);
+        vector::for_each_mut<PackageMetadata>(&mut registry.packages, |pack| {
```

Comment:
Can you deploy multiple packages to the same code object?

Reply:
As we discussed, current design allows the user to deploy one package to an object at a time. Later we will consider this additional feature if there is demand for this.